core/sql/src/main/java/org/trafodion/sql/HBaseClient.java

```diff
@@ -1395,6 +1422,169 @@ private boolean estimateRowCountBody(String tblName, int partialRowSize,
+      }
+
+      boolean retcode = true; 
+      rc[0] = 0;
+
+      HConnection connection = null;
+      HTableInterface table = null; 
+      connection = HConnectionManager.createConnection(config);
+      table = connection.getTable(tblName);
+
+      int putKVsSampled = 0;
+      int nonPutKVsSampled = 0;
+      int missingKVsCount = 0;
+      long totalEntries = 0;   // KeyValues in all HFiles for table
+      long totalSizeBytes = 0; // Size of all HFiles for table 
+
+      final int finalNumCols = numCols;
+
+      Batch.Call<TrxRegionService, TrafEstimateRowCountResponse> callable = 
+        new Batch.Call<TrxRegionService, TrafEstimateRowCountResponse>() {
+          ServerRpcController controller = new ServerRpcController();
+          BlockingRpcCallback<TrafEstimateRowCountResponse> rpcCallback = 
+            new BlockingRpcCallback<TrafEstimateRowCountResponse>();         
+
+          @Override
+          public TrafEstimateRowCountResponse call(TrxRegionService instance) throws IOException {    
+            if (logger.isDebugEnabled()) logger.debug("call method for TrxRegionService was called");
+            
+            // one of these God-awful long type identifiers common in Java/Maven environments...
+            org.apache.hadoop.hbase.coprocessor.transactional.generated.TrxRegionProtos.TrafEstimateRowCountRequest.Builder
+              builder = TrafEstimateRowCountRequest.newBuilder();        
+            builder.setNumCols(finalNumCols);
+        
+            instance.trafEstimateRowCount(controller, builder.build(), rpcCallback);
+            TrafEstimateRowCountResponse response = rpcCallback.get();
+            if (logger.isDebugEnabled()) {
+              if (response == null)
+                logger.debug("response was null");
+              else
+                logger.debug("response was non-null");
+              if (controller.failed())
+                logger.debug("controller.failed() is true");
+              else
+                logger.debug("controller.failed() is false");
+              if (controller.errorText() != null)
+                logger.debug("controller.errorText() is " + controller.errorText());
+              else
+                logger.debug("controller.errorText() is null");
+              IOException ioe = controller.getFailedOn();
+              if (ioe != null)
+                logger.debug("controller.getFailedOn() returned " + ioe.getMessage());
+              else
+                logger.debug("controller.getFailedOn() returned null");
+            }
+            return response;        
+          }
+      };
+    
+      Map<byte[], TrafEstimateRowCountResponse> result = null;
+      try {
+        result = table.coprocessorService(TrxRegionService.class, null, null, callable);
+      } catch (Throwable e) {
+        throw new IOException("Exception from coprocessorService caught in estimateRowCountViaCoprocessor",e);
```

Comment:
It is good to throw exception, but I think that the estimate earlier returned some default values if there is any issue with getting estimates.  It would be better to retain the same semantics to avoid surprises.